src/FixedDataTableRow.js

```diff
@@ -175,23 +182,36 @@ class FixedDataTableRowImpl extends React.Component {
      */
     template: PropTypes.oneOf(['cell', 'footer', 'header']).isRequired,
   };
+  constructor(props) {
+    super(props);
+    this._initialRender = true;
+  }
+
+  componentDidMount() {
+    this._initialRender = false;
+  }
 
   shouldComponentUpdate(nextProps) {
     // only skip updates while scrolling
     if (!nextProps.isScrolling) {
       return true;
     }
 
-    // if row is not visible then no need to render it
-    // change in visibility is handled by the parent
+    // if row's visibility has changed, then update it
+    if (this.props.visible !== nextProps.visible) {
+      return true;
+    }
+
+    // if row is still not visible then no need to update
     if (!nextProps.visible) {
       return false;
     }
 
-    // Only update the row if scrolling leads to a change in horizontal offsets.
-    // The vertical offset is taken care of by the wrapper
+    // if offsets haven't changed for the same row while scrolling, then skip update
     return !(
+      nextProps.isScrolling &&
       this.props.index === nextProps.index &&
+      this.props.offsetTop === nextProps.offsetTop &&
       this.props.scrollLeft === nextProps.scrollLeft
     );
   }
```

Comment:
These are the lifecyclmethods of react .We have to include all these methods into FixedDataTableRowImpl .